File: FunFinance.API/Models/Category.cs
using System.ComponentModel.DataAnnotations;

namespace FunFinance.API.Models
{
    public class Category
    {
        public int Id { get; set; }

        [Required, StringLength(50)]
        public required string Name { get; set; }

        [StringLength(200)]
        public string? Description { get; set; }

        // Зовнішній ключ до сім'ї
        public int FamilyId { get; set; }

        // Навігаційна властивість
        public required virtual Family Family { get; set; }
        public virtual ICollection<Expense> Expenses { get; set; } = new List<Expense>();
    }
}

File: funfinance-frontend/src/models/category.ts
export interface Category {
    id: number;
    name: string;
    description?: string;
    familyId: number;
    expenses?: Expense[];
  }
  
  export interface CategoryCreate {
    name: string;
    description?: string;
    familyId: number;
  }
  
  export interface CategoryUpdate {
    name?: string;
    description?: string;
  }
  
  // Імпортуємо Expense з моделі expense
  import { Expense } from './expense';